File: 1200/327A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> a(n);
    int one = 0; // count of 1s in the original array

    for (int i = 0; i < n; i++) {
        cin >> a[i];
        if (a[i] == 1) one++;

        // Transform the array:
        // - Replace 1 with -1 (flipping it would be a loss)
        // - Replace 0 with +1 (flipping it would be a gain)
        if (a[i] == 1) a[i] = -1;
        else a[i] = 1;
    }

    // Edge case: if all elements are 1, flipping any subarray reduces count
    if (one == n) {
        cout << one - 1 << "\n";
    }
    // Edge case: if all elements are 0, flipping the whole array gives all 1s
    else if (one == 0) {
        cout << n << "\n";
    }
    else {
        // Apply Kadane’s algorithm to find maximum subarray sum
        // This represents the maximum net gain achievable by flipping one segment
        int sum = INT_MIN;
        int s = 0;
        for (int i = 0; i < n; i++) {
            s = max(a[i], s + a[i]); // either start new subarray or extend previous
            sum = max(sum, s);       // track best sum so far
        }

        // Final answer = original count of 1s + best gain
        cout << one + sum << "\n";
    }

    return 0;
}

File: 1200/327A/main.java
import java.util.*;

public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        int a[] = new int[n];
        int one = 0;
        
        for(int i=0;i<n;i++)
        {
            a[i] = sc.nextInt();
            if(a[i] == 1)
                one++;

            // replacing 1 with -1
            // replacing 0 with +1
            
            if(a[i] == 1)
                a[i] = -1;
            else
                a[i] = 1;
        }
        
        if(one == n)
            System.out.println(one-1);
        else if(one == 0)
            System.out.println(n);
        else
        {
            // using Kadane's maximum sub array sum
            
            int sum = Integer.MIN_VALUE;
            int s = 0;
            for(int i=0;i<n;i++)
            {
                s = Math.max(a[i], s+a[i]);
                sum = Math.max(sum, s);
            }
            sum = Math.max(sum, s);
            
            System.out.println(one + sum);
        }
    }
}